### PR TITLE
fix a failure with multi-threaded repodata downloads

### PR DESCRIPTION
**Pull request by HugoTian**
Fix  issue #3069

### CODE REVIEW

**Comment by kalefranz** [comment]
@HugoTian I tagged you on another bug fix for #3060.  In general, I try to keep PRs for bug fixes with issues limited to fixing one bug.  So I would have split this into two separate PRs.  Let's just make this one for #3069.

**Comment by HugoTian** [comment]
@kalefranz , I pass all the CI test, and fails those coverage report. Is there any way to fix that ?

**Comment by kalefranz** [comment]
I'm not worried about the coverage report failures as long as the PR looks good.

File: conda/fetch.py [hunk 1]
```diff
@@ -305,6 +305,11 @@ def fetch_index(channel_urls, use_cache=False, unknown=False, index=None):
             futures = tuple(executor.submit(fetch_repodata, url, use_cache=use_cache,
                                             session=CondaSession()) for url in urls)
             repodatas = [(u, f.result()) for u, f in zip(urls, futures)]
+
+        except Exception as e:
```
Comment (kalefranz) [inline conda/fetch.py]: `except Exception` always sends off warning signals for me.  I'd prefer if we could narrow this down to the actually exception we're trying to catch.  I guess the one from the bug report this is attempting to fix.

also, please add a `log.debug()` here with info about the exception `e`.

Comment (kalefranz) [inline conda/fetch.py]: Actually, just above at line 298, someone has added this.

         # concurrent.futures is only available in Python >= 3.2 or if futures is installed
         # RuntimeError is thrown if number of threads are limited by OS

That's an awesome comment right after an `except`.  It's not always clear when an exception is expected.  That comment tells you exactly why we're catching both `ImportError` and `RuntimeError`.

Comment (HugoTian) [inline conda/fetch.py]: Great to learn that, and I will fix this.

**Comment by HugoTian** [comment]
Make a big git mistake, will create new branch for PR

**Comment by auberginekenobi** [comment]
@HugoTian ran into the same problem as described in #3069, using the same cluster system. Wonder if @yasheshgaur or others have reported the issue since?

**Comment by kalefranz** [comment]
@auberginekenobi Your issue will be resolved with https://github.com/conda/conda/pull/6653

**Pull request merged**